daemon/delete.go

```diff
@@ -141,6 +141,15 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 		}
 	}
 
+	// Ensure the health probe is stopped.
+	//
+	// If the contianer is continuously unhealthy, it may not
+	// happen automatically as the container would have exited
+	// before a call to `rm <-f>` which would not have lead to
+	// stop of the health monitor since the container was
+	// still marked as running
+	daemon.updateHealthMonitor(container)
```

Comment:
Just realized that `containerStop()` already stop the health check.

Going to go through the flow again, there's definitely a race there, but I'm thinking it's more in the `StateChanged()` daemon callback now.